src/NuGet.Core/NuGet.Build.Tasks/RestoreTaskEx.cs

```diff
@@ -253,33 +253,35 @@ internal string GetProcessFileName(string processFileName)
 #endif
         }
 
-        private Dictionary<string, string> GetGlobalProperties()
+        /// <summary>
+        /// Enumerates a list of global properties for the current MSBuild instance.
+        /// </summary>
+        /// <returns>An <see cref="IEnumerable{T}" /> of <see cref="KeyValuePair{TKey, TValue}" /> objects containing global properties.</returns>
+        internal IEnumerable<KeyValuePair<string, string>> GetGlobalProperties()
         {
+            IReadOnlyDictionary<string, string> globalProperties = null;
+
 #if IS_CORECLR
             // MSBuild 16.5 and above has a method to get the global properties, older versions do not
-            Dictionary<string, string> msBuildGlobalProperties = BuildEngine is IBuildEngine6 buildEngine6
-                ? buildEngine6.GetGlobalProperties().ToDictionary(i => i.Key, i => i.Value, StringComparer.OrdinalIgnoreCase)
-                : new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            if (BuildEngine is IBuildEngine6 buildEngine6)
+            {
+                globalProperties = buildEngine6.GetGlobalProperties();
+            }
 #else
-            var msBuildGlobalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-
             // MSBuild 16.5 added a new interface, IBuildEngine6, which has a GetGlobalProperties() method.  However, we compile against
             // Microsoft.Build.Framework version 4.0 when targeting .NET Framework, so reflection is required since type checking
             // can't be done at compile time
-            var buildEngine6Type = typeof(IBuildEngine).Assembly.GetType("Microsoft.Build.Framework.IBuildEngine6");
+            Type buildEngine6Type = typeof(IBuildEngine).Assembly.GetType("Microsoft.Build.Framework.IBuildEngine6");
 
             if (buildEngine6Type != null)
             {
-                var getGlobalPropertiesMethod = buildEngine6Type.GetMethod("GetGlobalProperties", BindingFlags.Instance | BindingFlags.Public);
+                MethodInfo getGlobalPropertiesMethod = buildEngine6Type.GetMethod("GetGlobalProperties", BindingFlags.Instance | BindingFlags.Public);
 
                 if (getGlobalPropertiesMethod != null)
                 {
                     try
                     {
-                        if (getGlobalPropertiesMethod.Invoke(BuildEngine, null) is IReadOnlyDictionary<string, string> globalProperties)
-                        {
-                            msBuildGlobalProperties.AddRange(globalProperties);
-                        }
+                        globalProperties = getGlobalPropertiesMethod.Invoke(BuildEngine, null) as IReadOnlyDictionary<string, string>;
```

Comment:
param name on null values to be more informative.

```suggestion
                        globalProperties = getGlobalPropertiesMethod.Invoke(BuildEngine, parameters: null) as IReadOnlyDictionary<string, string>;
```